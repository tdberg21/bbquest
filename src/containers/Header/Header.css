.header {
  background-color: #111;
  display: flex;
  width: 100%;
  justify-content: flex-end;

}

ul {
  display: inherit;
  list-style: none;
  margin-right: 2rem;
}

.nav-links {
  color: #F3F3F3;
  font-weight: 300;
  margin: .75rem;
  text-decoration: none;
}