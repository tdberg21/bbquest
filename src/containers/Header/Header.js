import React, { Component } from 'react';
import { NavLink, withRouter } from 'react-router-dom';


export default class Header extends Component {
  constructor () {
    super();

  }

  render() {
    return (
      <div className='header'>
        <ul className='navigation-links'>
          <li>
            <NavLink className='home-link' to='/'>
              Home
            </NavLink>
          </li>
          <li>
            <NavLink className='login-link' to='/login'>
               Login
            </NavLink>
          </li>
        </ul>
      </div>
    );
  }
}