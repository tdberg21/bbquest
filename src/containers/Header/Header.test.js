import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('HEADER TESTS', () => { 
  it('should match the snapshot without a user', () => {
    const mockUser = { };
    const mockVisited = [];
    const wrapper = shallow(<Header user={mockUser} visited={mockVisited} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with a user', () => {
    const mockUser = {username:'tman'};
    const mockVisited = [];
    const wrapper = shallow(<Header user={mockUser} visited={mockVisited}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with visited restaurants', () => {
    const mockUser = { username:'tman' };
    const mockVisited = ['peewees', 'moes'];
    const wrapper = shallow(<Header user={mockUser} visited={mockVisited}/>);

    expect(wrapper).toMatchSnapshot();
  });
});