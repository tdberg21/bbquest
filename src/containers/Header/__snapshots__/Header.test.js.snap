// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER TESTS should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="navigation-links"
>
        <li>
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="home-link"
                        to="/"
                >
                        Home
                </NavLink>
        </li>
        <li>
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="login-link"
                        to="/login"
                >
                        Login
                </NavLink>
        </li>
        <li>
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="search-link"
                        to="/search"
                >
                        Search
                </NavLink>
        </li>
</ul>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="home-link"
                        to="/"
            >
                        Home
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="login-link"
                        to="/login"
            >
                        Login
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="search-link"
                        to="/search"
            >
                        Search
            </NavLink>
</li>,
        ],
        "className": "navigation-links",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="home-link"
              to="/"
>
              Home
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Home",
              "className": "home-link",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="login-link"
              to="/login"
>
              Login
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Login",
              "className": "login-link",
              "to": "/login",
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="search-link"
              to="/search"
>
              Search
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Search",
              "className": "search-link",
              "to": "/search",
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="navigation-links"
>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="home-link"
                              to="/"
                    >
                              Home
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="login-link"
                              to="/login"
                    >
                              Login
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="search-link"
                              to="/search"
                    >
                              Search
                    </NavLink>
          </li>
</ul>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="home-link"
                            to="/"
              >
                            Home
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="login-link"
                            to="/login"
              >
                            Login
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="search-link"
                            to="/search"
              >
                            Search
              </NavLink>
</li>,
          ],
          "className": "navigation-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="home-link"
                to="/"
>
                Home
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Home",
                "className": "home-link",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="login-link"
                to="/login"
>
                Login
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Login",
                "className": "login-link",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="search-link"
                to="/search"
>
                Search
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Search",
                "className": "search-link",
                "to": "/search",
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
